src/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandHandler.cs

```diff
@@ -123,6 +125,46 @@ public async Task Handle(MigrateRepoCommandArgs args)
         }
     }
 
+    private void ValidateBbsSharedHome(MigrateRepoCommandArgs args)
+    {
+        if (!string.IsNullOrEmpty(args.BbsSharedHome))
+        {
+            if (IsRunningOnBitbucketServer())
+            {
+                if (!Directory.Exists(args.BbsSharedHome))
+                {
+                    throw new OctoshiftCliException($"Invalid --bbs-shared-home path: '{args.BbsSharedHome}'. Directory does not exist.");
+                }
+            }
+            else
+            {
+                if (!Directory.Exists(args.BbsSharedHome))
+                {
+                    throw new OctoshiftCliException($"Invalid --bbs-shared-home path: '{args.BbsSharedHome}'. Directory does not exist.");
+                }
+            }
+        }
+    }
+
+    private void ValidateArchivePath(MigrateRepoCommandArgs args)
+    {
+        if (!string.IsNullOrEmpty(args.ArchivePath))
+        {
+            if (!File.Exists(args.ArchivePath))
+            {
+                throw new OctoshiftCliException($"Invalid --archive-path: '{args.ArchivePath}'. File does not exist.");
+            }
+            _log.LogInformation($"Archive path for upload: {args.ArchivePath}");
+        }
+    }
```

Comment:
A few notes here too:
1. This check also needs to go under `ValidateUploadOptions()` method.
2. Instead of `File.Exists`, the `_fileSystemProvider.FileExists()` abstraction should be used. This will simplify unit testing.
3. There is no need to add an extra info log as all args will get logged automatically.